ansible/roles/machine/provision/tasks/main.yml

```diff
@@ -103,6 +103,30 @@
     state: enforcing
   when: selinux_enforcing is defined and selinux_enforcing
 
+- block:
+    - name: install FIPS dependencies
+      dnf:
+        state: latest
+        name:
+          - fips-mode-setup
+
+    - name: run fips-mode-setup
+      shell: fips-mode-setup --enable
+
+    - name: reboot the machine
+      reboot:
```

Comment:
There's a problem here. Vagrant is not able to detect a reboot, so it loses its synced folder functionality, that means we can't collect the artifacts once test is done.

Is the reboot really necessary?

Reply:
@netoarmando first of all, thank you for your review!

The `reboot` is necessary, as FIPS setup includes changing the boot parameters. Without the reboot, ` fips-mode-setup --is-enabled` returns return code 2, which means disabled.

Regarding possible solution, I think, we can either:
- execute some sort of re-sync command after reboot (`reboot` has a hook for that)
- move the reboot to provisioning phase before the synchronization is established